plugins/guests/suse/cap/change_host_name.rb

```diff
@@ -4,19 +4,38 @@ module Cap
       class ChangeHostName
         def self.change_host_name(machine, name)
           comm = machine.communicate
-
           basename = name.split(".", 2)[0]
-          if !comm.test('test "$(hostnamectl --static status)" = "#{basename}"', sudo: false)
+          if !comm.test('test "$(hostname)" = "#{basename}"', sudo: false)
             comm.sudo <<-EOH.gsub(/^ {14}/, '')
-              hostnamectl set-hostname '#{basename}'
-
-              # Prepend ourselves to /etc/hosts
-              grep -w '#{name}' /etc/hosts || {
-                sed -i'' '1i 127.0.0.1\\t#{name}\\t#{basename}' /etc/hosts
-              }
+            hostname #{name}
+            echo #{name} > /etc/HOSTNAME
             EOH
+            network_with_hostname = machine.config.vm.networks.map {|t, c| c if c[:hostname] }.compact[0]
+            if network_with_hostname
+              replace_host(comm, name, basename, network_with_hostname[:ip])
+            else
+              add_hostname_to_loopback(comm, name, basename)
+            end
           end
         end
+
+        def self.add_hostname_to_loopback(comm, name, basename)
+          # Add hostname to /etc/hosts if not already there
+          comm.sudo <<-EOH.gsub(/^ {14}/, '')
+          grep -w '#{name}' /etc/hosts || {
+            sed -i'' '1i 127.0.0.1\\t#{name}\\t#{basename}' /etc/hosts
```

Comment:
When doing the default add to loopback, we want to use an unused loopback address. Debian does this using `127.0.1.1` which is reasonable but we'll want to check the `/etc/hosts` file and confirm it's not already in use, and if it is, increment until an unused one is found.